docs/src/main/asciidoc/spanner.adoc

```diff
@@ -0,0 +1,190 @@
+|===
+
+==== Relationships
+
+The dialect supports all of the standard relational annotations:
+
+- `@OneToOne`
+- `@OneToMany`
+- `@ManyToOne`
+- `@ManyToMany`
+
+These can be used via `@JoinTable` or `@JoinColumn`.
+However, because Cloud Spanner does not support Foreign Keys, foreign-key-columns are just regular columns in Cloud Spanner.
+
+NOTE: The lack of foreign keys also means database-side cascading deletes are not supported via the `@OnDelete(action = OnDeleteAction.CASCADE)` annotation because there is no `ON DELETE CASCADE` constraint in Cloud Spanner DDL.
+However, Hibernate-side cascading operations such as `@ManyToOne(cascade = {CascadeType.ALL})` are supported.
+
+
+==== Constraints
+
+Cloud Spanner does not support constraints.
+As a result, `SpannerDialect` does not currently support any constraints such as `FOREIGN KEY`, `UNIQUE`, or `ON DELETE CASCADE`.
+
+NOTE: Hibernate doesn't directly rely on the existence of constraints to perform its operations and leaves the enforcement relationship links to the database.
+
+The lack of the foreign key constraint affects relationships and collection properties annotated with `@ElementCollection`.
+
+
+==== Generated IDs
+
+Hibernate's `@GeneratedValue` annotation for `long` properties is supported but not recommended:
+[source, java]
+----
+@Entity
+public class Employee {
+
+  @Id
+  @GeneratedValue
+  public Long id;
+}
+----
+
+This results in sequential IDs that are https://cloud.google.com/spanner/docs/schema-and-data-model#primary_keys[not optimal for Cloud Spanner] and requires locking of the `hibernate_sequence` table for inserts.
+
+You should use locally-generated UUID key values instead:
+[source, java]
+----
+@Entity
+public class Employee {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  @Type(type="uuid-char")
+  public UUID id;
+}
+----
+
+The `@Type(type="uuid-char")` annotation specifies that this UUID value will be stored in Cloud Spanner as a `STRING` column.
+Leaving out this annotation causes a `BYTES` column to be used.
+
+
+==== Table Catalogs and Schemas
+
+Cloud Spanner does not support table names with Catalog and Schema components:
+
+[source, java]
+----
+// Not supported: `public.store.book` is not a valid Cloud Spanner table name.
+@Table(
+  catalog = "public",
+  schema = "store",
+  name = "book"
+)
+
+// Supported.
+@Table(
+  name = "book"
+)
+----
+
+
+==== Column Default Values
+
+The dialect does not currently set default values based on the https://docs.jboss.org/hibernate/orm/5.4/javadocs/org/hibernate/annotations/ColumnDefault.html[`@ColumnDefault`] annotation,
+because `null` values aren't specially handled and are stored just like other values by Cloud Spanner and its driver.
+
+
+==== Decimal and Numeric Types
+
+Cloud Spanner does not provide native support for https://cloud.google.com/spanner/docs/storing-numeric-data[arbitrary-precision decimal numbers], but does provide some alternative approaches.
```

Comment:
If we're going to mention alternative approaches, then we have to describe them. Otherwise, let's just omit that part.

```suggestion
Cloud Spanner does not provide native support for https://cloud.google.com/spanner/docs/storing-numeric-data[arbitrary-precision decimal numbers], such as `NUMERIC` and `DECIMAL`.
```